nav2_controller/src/controller_server.cpp

```diff
@@ -464,7 +499,7 @@ void ControllerServer::computeAndPublishVelocity()
       nav_2d_utils::twist2Dto3D(twist),
       goal_checkers_[current_goal_checker_].get());
     last_valid_cmd_time_ = now();
-  } catch (nav2_core::PlannerException & e) {
+  } catch (nav2_core::NoValidControl & e) {
```

Comment:
This is subtle - add a comment here: "Only no valid control exception types are valid to attempt to have control patience, as other types will not be resolved with more attempts."